src/OpenTelemetry.Resources.AssemblyMetadata/AssemblyMetadataResourceBuilderExtensions.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using OpenTelemetry.Internal;
+using OpenTelemetry.Resources.AssemblyMetadata;
+
+namespace OpenTelemetry.Resources;
+
+/// <summary>
+/// Extension methods to simplify registering the assembly metadata resource detector.
+/// </summary>
+public static class AssemblyMetadataResourceBuilderExtensions
+{
+    /// <summary>
+    /// Enables the assembly metadata resource detector.
+    /// </summary>
+    /// <param name="builder">The <see cref="ResourceBuilder"/> being configured.</param>
+    /// <returns>The instance of <see cref="ResourceBuilder"/> being configured.</returns>
+    /// <inheritdoc cref="AssemblyMetadataDetector" />
+    public static ResourceBuilder AddAssemblyMetadataDetector(this ResourceBuilder builder)
```

Comment:
As noted in the other file, I suggest adding an extension to support the assemblies being specified, with the entry assembly as the default behaviour.

Whether that's added or not, that behaviour should be included in the XML documentation so someone doesn't need to look at the code to determine how it chooses the assemblies.